Cargo.toml

```diff
@@ -40,6 +40,47 @@ default-members = [
     "libparsec/crates/types",
 ]
 
+[workspace.dependencies]
+anyhow = { version = "1.0.71", default-features = false, features = ["std"] }
```

Comment:
What is the rational for setting `default-feature` and `features` here or in the crate ?

(I would say we want to only list dependency version here, so every one of them should be in `default-feature = false`)

Reply:
For me the rational would be:

- Always set `default-features` to `false` and list the `features` you actually rely on
  
  I haven't done it everywhere, because we surely discuss about that.

  The goal of `default-features` is to be explicit on the features we use / rely on.
  If a dependency add a feature in the future it won't be enabled by default.

- `features` in `workspace.dependencies` are for the `features` that are required across all the workspace members.

Reply:
>  Always set default-features to false and list the features you actually rely on
> [...]
> If a dependency add a feature in the future it won't be enabled by default.

I think this is overkill given modifying default features [is considered a breaking change](https://doc.rust-lang.org/cargo/reference/semver.html#major-removing-a-feature-from-a-feature-list-if-that-changes-functionality-or-public-items)

> features in workspace.dependencies are for the features that are required across all the workspace members.

After grepping the code base for dependencies with features:
```
2x tokio = { version = "1.28", features = ["rt-multi-thread"] }
tokio = { version = "1.28.0", features = ["rt", "time", "sync", "macros"] }
tokio = { version = "1.28.1", features = ["rt", "time", "sync", "macros"] }
tokio = { version = "1.28.1", features = ["full"] }
tokio = { version = "1.28.1", features = ["fs", "sync"] }
tokio = { version = "1.28.0", features = ["fs", "sync"] }
tokio = { version = "1.28.0", features = ["fs", "sync"] }
tokio = { version = "1.28", features = ["rt", "rt-multi-thread"] }

4x serde = { version = "1.0.147", features = ["derive"] }

3x libsqlite3-sys = { version = "0.25.2", features = ["bundled"] }

3x bytes = { version = "1.4.0", features = ["serde"] }

uuid = { version = "1.2.2", features = ["v4", "fast-rng"] }
2x uuid = { version = "1.2.2", features = ["serde", "v4", "fast-rng"] }

2x pretty_assertions = { version = "1.3.0", features = ["unstable"] }

reqwest = { version = "0.11.17", features = ["native-tls-vendored"] }
reqwest = { version = "0.11.17", default-features = false }

neon = { version = "0.10", default-features = false, features = ["napi-6", "channel-api", "promise-api"] }
hyper = { version = "0.14.26", features = ["server"] }
openssl = { version = "0.10.48", optional = true, features = ["vendored"] }
zeroize = { version = "1.6.0", optional = true, features = ["alloc"]}
crypto_box = { version = "0.8.2", features = ["std"] }  # Optional ruscrypto dep
generic-array = { version = "0.14.7", features = ["serde"] }  # Optional ruscrypto dep
getrandom_01 = { package = "getrandom", version = "0.1.16", features = ["wasm-bindgen"] }  # Optional rustcrypto dep
getrandom_02 = { package = "getrandom", version = "0.2.3", features = ["js"] }  # Optional rustcrypto dep
web-sys = { version = "0.3.60", features = ["Window"] }
web-sys = { version = "0.3.60", features = ["Window", "Storage"] }
instant = { version = "0.1.12", features = ["wasm-bindgen"] }
diesel = { version = "2.0.3", features = ["sqlite", "returning_clauses_for_sqlite_3_35"], default-features = false }
anyhow = { version="1.0.71", features = ["backtrace"] }
pyo3 = { version = "0.17.3", features = ["multiple-pymethods", "extension-module"] }
```

I don't think there is a clear way to define feature required across the workspace.
Just like there is no dependency  needed across the workspace, but some crate need some dependency to achieve some goal (hence also needing some features is such dependency).

In the end I'd rather keep things simple so that adding a new dependency is obvious enough (==> workspace depencies specify version, features are specified on per-crate dependencies).

Reply:
> workspace depencies specify version, features are specified on per-crate dependencies).

Some dependencies always use the same set of `features` (see `serde` with `derive`, `web-sys` with `Windows`)

Reply:
> Some dependencies always use the same set of features (see serde with derive, web-sys with Windows)

Yes, but there is no hard reason for that, it just happen to be the case. So I don't see the gain in merging the features: in fact this would make harder to spot when a feature is no longer needed.

Reply:
done